src/lib/coda_base/new_account_payload.ml

```diff
@@ -0,0 +1,40 @@
+open Core_kernel
+open Import
+
+[%%versioned
+module Stable = struct
+  module V1 = struct
+    type t =
+      { token_id: Token_id.Stable.V1.t
+      ; token_owner_pk: Public_key.Compressed.Stable.V1.t
+      ; receiver_pk: Public_key.Compressed.Stable.V1.t }
+    [@@deriving compare, eq, sexp, hash, yojson]
+
+    let to_latest = Fn.id
+  end
+end]
+
+type t = Stable.Latest.t =
+  { token_id: Token_id.t
+  ; token_owner_pk: Public_key.Compressed.t
+  ; receiver_pk: Public_key.Compressed.t }
+[@@deriving compare, eq, sexp, hash, yojson]
+
+let receiver_pk {receiver_pk; _} = receiver_pk
+
+let receiver {token_id; receiver_pk; _} =
+  Account_id.create receiver_pk token_id
+
+let source_pk {token_owner_pk; _} = token_owner_pk
+
+let source {token_id; token_owner_pk; _} =
+  Account_id.create token_owner_pk token_id
+
+let token {token_id; _} = token_id
+
+let gen =
+  let open Quickcheck.Generator.Let_syntax in
+  let%bind token_id = Token_id.gen_non_default in
+  let%bind token_owner_pk = Public_key.Compressed.gen in
+  let%map receiver_pk = Public_key.Compressed.gen in
```

Comment:
I think `token_owner_pk` will always be the same as `receiver_pk`, because the generator is deterministic. Maybe use seeds to get distinct values?

Reply:
I'm wrong.